src/Data/Text.hs

```diff
@@ -1881,6 +1883,43 @@ isInfixOf needle haystack
     | otherwise          = not . L.null . indices needle $ haystack
 {-# INLINE [1] isInfixOf #-}
 
+-- 2021-09-29: NOTE:
+--  * after the implementation - determine & mention the big O
+-- | The 'isSubsequenceOf' function takes the main text and the subsequnce
+-- to find and returns 'True' iff the second argument is a subsequence
+-- of the first.
+--
+-- "Subsequence" used in the meaning of: characters of the second argument
+-- appear in same sequential order in the main data, to say second argument can
+-- be derived by deleting some (any) or no elements from the first.
+--
+-- Examples:
+--
+-- >>> isSubsequenceOf "1234567" "1356"
+-- True
+--
+-- >>> isSubsequenceOf "1234567" "21"
+-- False
+--
+-- `isSubsequenceOf` is the base case & implementation of fuzzy search.
+isSubsequenceOf :: Text -> Text -> Bool
+isSubsequenceOf tf sf
+  | length sf > length tf = False
+  | otherwise = subseqOf tf sf
+ where
+  subseqOf :: Text -> Text -> Bool
+  subseqOf t s =
+    on f uncons t s
+   where
+    f :: Maybe (Char, Text) -> Maybe (Char, Text) -> Bool
+    f _ Nothing = True
+    f Nothing _ = False
+    f (Just (tc,ts)) (Just (sc,ss)) =
+      subseqOf ts $
+        if tc == sc
+          then s
+          else ss
```

Comment:
Is there a good reason not to just use 

```haskell
isSubsequenceOf needles haystack = List.isSubsequenceOf (T.unpack needles) (T.unpack haystack)
```
This code is doing exactly the same work, since `T.unpack` (as far as I understand) is just repeated calls to `uncons`. I can't see how there's be any performance here (though I could see an argument to using an optimised implementation in C if there's a more efficient algorithm that takes advantage of the UTF-8 encoding).